docs/src/user_docs/stdlib/math/u64.md

```diff
@@ -0,0 +1,69 @@
+
+## Arithmetic operations
+
+| Procedure   | Description   |
+| ----------- | ------------- |
+| overflowing_add |  Performs addition of two unsigned 64 bit integers preserving the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [overflowing_flag, c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| wrapping_add |  Performs addition of two unsigned 64 bit integers discarding the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| checked_add |  Performs addition of two unsigned 64 bit integers, fails when overflowing.<br /> The input values are assumed to be represented using 32 bit limbs, fails if they are not.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| wrapping_sub |  Performs subtraction of two unsigned 64 bit integers discarding the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a - b) % 2^64 |
+| checked_sub |  Performs subtraction of two unsigned 64 bit integers, fails when underflowing.<br /> The input values are assumed to be represented using 32 bit limbs, fails if they are not.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a - b) % 2^64 |
+| overflowing_sub |  Performs subtraction of two unsigned 64 bit integers preserving the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [underflowing_flag, c_hi, c_lo, ...], where c = (a - b) % 2^64 |
+| wrapping_mul |  Performs multiplication of two unsigned 64 bit integers discarding the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a * b) % 2^64 |
+| overflowing_mul |  Performs multiplication of two unsigned 64 bit integers preserving the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_mid_hi, c_mid_lo, c_lo, ...], where c = (a * b) % 2^64<br /> This takes 18 cycles. |
```

Comment:
I think the naming of the outputs could be clearer here as well for communicating the overflow (overflowing_mul)